Sources/NIOSOCKS/Channel Handlers/SOCKSClientHandler.swift

```diff
@@ -165,4 +162,42 @@ extension SOCKSClientHandler {
         context.writeAndFlush(self.wrapOutboundOut(buffer), promise: nil)
     }
     
+    private func emptyInboundAndOutboundBuffer(context: ChannelHandlerContext) {
+        if let inboundBuffer = self.inboundBuffer, inboundBuffer.readableBytes > 0 {
+            // after the SOCKS handshake message we already received further bytes.
+            // so let's send them down the pipe
+            let data = self.wrapInboundOut(self.inboundBuffer!)
```

Comment:
We don't need a force unwrap here, we have done a conditional binding. Also, we can move this wrap into the call to `fireChannelRead`.

Reply:
Urgh. Oversight. Fixed.